pkg/tenant/projects.go

```diff
@@ -10,29 +10,68 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+// TenantProjectList retrieves all projects assigned to a tenant
+// and returns a 200 OK response.
+//
+// Route: //tenant/:tenantID/projects
 func (s *Server) TenantProjectList(c *gin.Context) {
-	c.JSON(http.StatusNotImplemented, "not implemented yet")
+	var (
+		err error
+	)
+	// Get the project's tenant ID from the URL and return a 400 response
+	// if the tenant ID is not a ULID.
+	var tenantID ulid.ULID
+	if tenantID, err = ulid.Parse(c.Param("tenantID")); err != nil {
+		log.Error().Err(err).Msg("could not parse tenant ulid")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse tenant ulid"))
+		return
+	}
+
+	// Get projects from the database and return a 500 response
+	// if not successful.
+	if _, err := db.ListProjects(c.Request.Context(), tenantID); err != nil {
+		log.Error().Err(err).Msg("could not fetch projects from the database")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not fetch projects from the database"))
+		return
+	}
+
+	// Build the response.
+	out := &api.TenantProjectPage{TenantProjects: make([]*api.Project, 0)}
+
+	tenantProject := &api.Project{}
+
+	out.TenantProjects = append(out.TenantProjects, tenantProject)
+
+	c.JSON(http.StatusOK, out)
 }
 
 func (s *Server) TenantProjectCreate(c *gin.Context) {
 	c.JSON(http.StatusNotImplemented, "not implemented yet")
 }
 
+// ProjectList retrieves all projects assigned to a tenant
+// and returns a 200 OK response.
+//
+// Route: /projects
 func (s *Server) ProjectList(c *gin.Context) {
-	// The following TODO task items will need to be
-	// implemented for each endpoint.
+	// TODO: Fetch the project's tenant ID from key.
 
-	// TODO: Add authentication and authorization middleware
-	// TODO: Identify top-level info
-	// TODO: Parse and validate user input
-	// TODO: Perform work on the request, e.g. database interactions,
-	// sending notifications, accessing other services, etc.
+	// Get projects from the database and return a 500 response
+	// if not successful.
+	if _, err := db.ListProjects(c.Request.Context(), ulid.ULID{}); err != nil {
```

Comment:
As noted above, I think we want to capture the db values passed back into a variable here which can be assigned to an attribute of `out`